### PR TITLE
Continue download after early end

### PR DESCRIPTION
**Pull request by fent**
Continuing from #128, this is mainly adding tests.

I also tested this with a real download, by pausing a download halfway for about 6 minutes, I was able to consistently get an ECONNRESET error and then continue the download.

@TimeForANinja please look over this, and feel free comment on anything I might have overlook.

### CODE REVIEW

File: example/range.js [hunk 1]
```diff
@@ -5,10 +5,10 @@ var ytdl = require('..');
 var url = 'https://www.youtube.com/watch?v=CehAKQL463M';
 var output = path.resolve(__dirname, 'video.mp4');
 
-var video = ytdl(url, { range: '0-1000' });
+var video = ytdl(url, { range: { start: 0, end: 1000 } });
 video.pipe(fs.createWriteStream(output));
 
 video.on('end', function() {
-  ytdl(url, { range: '1001-' })
+  ytdl(url, { range: { start: 0 } })
```
Comment (TimeForANinja) [inline example/range.js]: Isn't this suppost to be 1000?

File: lib/index.js [hunk 2]
```diff
@@ -79,30 +83,48 @@ var redirectCodes = new Set([301, 302, 303, 307]);
...
       myres.unpipe();
+      myres.destroy();
     }
     stream.emit('abort');
   };
 
   req.on('error', function(err) {
-    stream.emit('error', err);
+    if (stream._isDestroyed || !myres) {
+      stream.emit('error', err);
```
Comment (TimeForANinja) [inline lib/index.js]: Not sure wether u shouldnt close the stream anyway but since i don't know better...

Comment (fent) [inline lib/index.js]: The stream should close itself after this.

**Comment by fent** [comment]
Thanks for the original PR again, and for reviewing.

**Pull request merged**